### PR TITLE
EPMCUXDA-7057: Rings pagination

### PR DESCRIPTION
**Pull request by SerafimovichEugene**
### What does this PR do?

Search page params for Rings were introduced.

### Jira ticket related to this PR

- [EPMCUXDA-7057](https://jira.epam.com/jira/browse/EPMCUXDA-7057)

#### Type of change

`What types of changes does your code introduce? Put an 'x' in all the boxes that apply:`

- [x] Adds new features.
- [x] Changes existing functionality.
- [ ] Deprecates features.
- [ ] Removes features.
- [ ] Fixes any bug.
- [ ] Other: security, build process, infrastructure, tests, docs, etc.

### CODE REVIEW

File: src/services/page-service.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+const { pageNumberDefault, pageSizeDefault } = config.get('paging');
+
+interface FindOptions<T> extends FindManyOptions<T> {
+  number: number;
+  size: number;
+}
+
+export const parsePageParams = <T>(query: Search): FindOptions<T> => {
+  const { pageNumber, pageSize, sortingColumn, sortingDirection = 'ASC' } = query;
```
Comment (mishelen) [inline src/services/page-service.ts]: just for consistency you could move `ASC` to `config` and desctructure with renaming and assigning defaults to `number` and `size` in the current statement

File: src/entities/common-interfaces.ts [hunk 2]
```diff
@@ -46,4 +46,17 @@ export interface AbleToExportAndImportEuring {
   importEURING(code: string): any;
 }
 
+export enum SortingDirection {
```
Comment (mishelen) [inline src/entities/common-interfaces.ts]: we seem to be contaminating the space of this namespace, which was originally intended for models. 

I think that the service itself can export the interfaces for these methods in addition to the methods.

File: src/controllers/rings-by-controller.ts [hunk 3]
```diff
@@ -21,10 +30,22 @@ export default class RingsByController extends AbstractController {
...
+    @QueryParam('sortingColumn') sortingColumn?: string,
+  ): Promise<RingListResponse> {
+    const paramsSearch = parsePageParams({ pageNumber, pageSize, sortingColumn, sortingDirection });
+    const [rings, totalElements] = await this.rings.findAndCount(paramsSearch);
+    return {
+      content: rings,
+      pageNumber: paramsSearch.number,
+      pageSize: paramsSearch.size,
+      totalElements,
+    };
```
Comment (mishelen) [inline src/controllers/rings-by-controller.ts]: it's a little weird to put `longVariable` in function to then get `shortVariable`  back to then remap it in `longVariable`

File: src/controllers/rings-by-controller.ts [hunk 4]
```diff
@@ -21,10 +30,22 @@ export default class RingsByController extends AbstractController {
 
   @GET
   @Path('/')
-  @Response<Ring[]>(200, 'List of all available rings.')
+  @Response<RingListResponse>(200, 'List of all available rings.')
```
Comment (mishelen) [inline src/controllers/rings-by-controller.ts]: Please add annotations for each endpoint

Comment (SerafimovichEugene) [reply src/controllers/rings-by-controller.ts]: what kind of annotations did you mean?

**Pull request merged**